### PR TITLE
scan: implement data source page requests and revisions tables ui

### LINKED ISSUE #670: Implement data source page requests and revisions tables UI

### PR DESCRIPTION
**Pull request by prin-r**
Fix #670

### CODE REVIEW

File: scan/src/pages/DataSourceIndexPage.re [hunk 1]
```diff
@@ -102,8 +102,8 @@ let make = (~dataSourceID, ~hashtag: Route.data_source_tab_t) => {
       {switch (hashtag) {
        | DataSourceExecute => <DataSourceExecute />
        | DataSourceCode => <DataSourceCode />
-       | DataSourceRequests => <div> {"Requests" |> React.string} </div>
-       | DataSourceRevisions => <div> {"Revisions" |> React.string} </div>
+       | DataSourceRequests => <DataSourceTable />
```
Comment (evilpeach) [inline scan/src/pages/DataSourceIndexPage.re]: Let's name it to `DataSourceRequestTable`

File: scan/src/components/RevisionTable.re [hunk 2]
```diff
@@ -0,0 +1,170 @@
...
+    style([
+      width(`percent(100.)),
+      boxShadow(Shadow.box(~x=`px(0), ~y=`px(2), ~blur=`px(4), Css.rgba(0, 0, 0, 0.08))),
+      backgroundColor(white),
+      display(`flex),
+      justifyContent(`center),
+      alignItems(`center),
+      height(`px(30)),
+      cursor(`pointer),
+    ]);
```
Comment (evilpeach) [inline scan/src/components/RevisionTable.re]: Remove unused styles

File: scan/src/components/data-source/DataSourceTable.re [hunk 3]
```diff
@@ -0,0 +1,219 @@
...
+    style([
+      width(`percent(100.)),
+      boxShadow(Shadow.box(~x=`px(0), ~y=`px(2), ~blur=`px(4), Css.rgba(0, 0, 0, 0.08))),
+      backgroundColor(white),
+      display(`flex),
+      justifyContent(`center),
+      alignItems(`center),
+      height(`px(30)),
+      cursor(`pointer),
+    ]);
```
Comment (evilpeach) [inline scan/src/components/data-source/DataSourceTable.re]: remove this as well

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — DISMISSED** [review]
LGTM

**Review by evilpeach — DISMISSED** [review]

**Comment by evilpeach** [comment]
@prin-r Please fix conflict

**Review by evilpeach — DISMISSED** [review]
LGTM

**Review by evilpeach — APPROVED** [review]

**Review by taobun — APPROVED** [review]

**Pull request merged**